tests/tdastro/astro_utils/test_passbands.py

```diff
@@ -0,0 +1,198 @@
+            assert passbands._download_transmission_table("test-band", "mock_file.dat")
+
+
+def test_passbands_download_transmission_table_http_error():
+    """Test that we handle HTTPError when downloading a transmission table."""
+    passbands = Passbands(bands=["test-band"])
+    with patch(
+        "urllib.request.urlretrieve",
+        side_effect=urllib.error.HTTPError(url="", code=404, msg="Not Found", hdrs=None, fp=None),
+    ):
+        assert not passbands._download_transmission_table("test-band", "mock_file.dat")
+
+
+def test_passbands_download_transmission_table_url_error():
+    """Test that we handle URLError when downloading a transmission table."""
+    passbands = Passbands(bands=["test-band"])
+    with patch("urllib.request.urlretrieve", side_effect=urllib.error.URLError("No host")):
+        assert not passbands._download_transmission_table("test-band", "mock_file.dat")
+
+
+def test_passbands_load_transmission_table_value_error():
+    """Test that we handle ValueError when loading a transmission table."""
+    passbands = Passbands(bands=["test-band"])
+    with patch("numpy.loadtxt", side_effect=ValueError("Mocked ValueError")):
+        assert not passbands._load_transmission_table("test-band", "mock_file.dat")
+
+
+def test_passbands_load_transmission_table_os_error():
+    """Test that we handle OSError when loading a transmission table."""
+    passbands = Passbands(bands=["test-band"])
+    with patch("numpy.loadtxt", side_effect=OSError("Mocked OSError")):
+        assert not passbands._load_transmission_table("test-band", "mock_file.dat")
+
+
+def test_passbands_load_all_transmission_tables(tmp_path):
+    """Test that we can download and load all transmission tables."""
+    band_ids = ["u", "g"]
+    passbands = Passbands(bands=band_ids)
+
+    # Mock the _get_data_path method
+    with patch.object(passbands, "_get_data_path"):
+        passbands.data_path = os.path.join(tmp_path, "band_data/")
+
+        # Mock the _download_transmission_table and _load_transmission_table methods
+        with patch.object(passbands, "_download_transmission_table", return_value=True), patch.object(
+            passbands, "_load_transmission_table", return_value=True
+        ):
+            passbands.load_all_transmission_tables()
+
+            # Assert _download_transmission_table was called for each band
+            assert passbands._download_transmission_table.call_count == len(band_ids)
+
+            # Assert _load_transmission_table was called for each band
+            assert passbands._load_transmission_table.call_count == len(band_ids)
+
+
+def test_passbands_phi_b():
+    """Test that we can calculate the value of phi_b for all wavelengths in a transmission table."""
+    passbands = Passbands()
+
+    u_band_transmission_table = np.array([[100.0, 0.5], [200.0, 0.75], [300.0, 0.25]])
+    g_band_transmission_table = np.array([[100.0, 0.75], [200.0, 0.5], [300.0, 0.25]])
```

Comment:
Given that these are just three points, can you manually compute the expected values of the normalized tables and compute the entries?